src/core/string_map.h

```diff
@@ -62,6 +62,13 @@ class StringMap : public DenseSet {
       return BreakToPair(ptr);
     }
 
+    void ReallocIfNeeded(float ratio) {
```

Comment:
You can document that this function requires that the iterator will be valid and omit/assert `curr_entry_->IsObject())` condition.

Reply:
I can't call SetObject() on a link node, that's why I check

Reply:
what do I miss? Should not it be the same like `operator*()`,  therefore it should never be `IsLink` ?